components/google_sheets/actions/add-single-row/add-single-row.mjs

```diff
@@ -32,15 +66,64 @@ export default {
         }),
       ],
     },
-    cells: {
-      propDefinition: [
-        googleSheets,
-        "cells",
-      ],
-    },
+    dataFormat,
+  },
+  async additionalProps() {
+    if (this.dataFormat == "column") {
+      if (!this.header) {
+        const rv = {};
+        rv.header = headerProp;
+        return rv;
+      } else {
+        if (this.header === "hasHeaders") {
+          const rv = {};
+          rv.header = headerProp;
+          const { values } = await this.googleSheets.getSpreadsheetValues(this.sheetId, `${this.sheetName}!1:1`);
```

Comment:
Any chance of values being null or do not have position 0? If so, could you please verify it before accessing `values[0]`?